src/main/java/reactor/netty/http/HttpOperations.java

```diff
@@ -297,15 +297,32 @@ protected final boolean markSentHeaderAndBody() {
 	 * @return a normalized uri without the leading and trailing '/' if present
 	 */
 	public static String resolvePath(String uri) {
-		String path = URI.create(uri).getPath();
+		if (uri.isEmpty()) {
+			return uri;
+		}
+
+		String path;
+		if (uri.charAt(0) == '/') {
+			path = uri;
+			for (int i = 0; i < path.length(); i++) {
+				char c = path.charAt(i);
+				if (c == '?' || c == '#') {
+					path = path.substring(0, i);
+					break;
+				}
+			}
+		}
+		else {
+			path = URI.create(uri).getPath();
+		}
```

Comment:
The if does not decode while the else does, which seems inconsistent:
```java
assertEquals("a b", HttpOperations.resolvePath("http://localhost:8080/a%20b"));
assertEquals("a%20b", HttpOperations.resolvePath("/a%20b"));
```

One option might be something like this instead?
```java
uri = (uri.charAt(0) == '/' ? "http://abc.org" + uri : uri); 
String path = URI.create(uri).getPath();
```